### PR TITLE
Add autorization header for transfer files

### PR DESCRIPTION
**Pull request by vbauzysvmware**
Fix for https://github.com/terraform-providers/terraform-provider-vcd/issues/227

Due to custom implementation, transfer files functionality wasn't assigning  authorization request header. vCD doesn't require that and works, but for users who are behind proxy/firewalls which filter out request without authorization request header upload isn't working. This fix reuse are common request building function, logging and request handling.

### CODE REVIEW

File: govcd/upload.go [hunk 1]
```diff
@@ -105,19 +105,22 @@ func uploadFile(client *Client, filePath string, uDetails uploadDetails) (int64,
...
 	util.Logger.Printf("[TRACE] Creating file upload request: %s, %v, %v, %v \n", requestUrl, offset, filePartSize, fileSizeToUpload)
 
-	uploadReq, err := http.NewRequest(http.MethodPut, requestUrl, bytes.NewReader(filePart))
+	parsedRequestURL, err := url.ParseRequestURI(requestUrl)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error decoding vdc response: %s", err)
 	}
 
+	uploadReq := client.NewRequestWitNotEncodedParams(nil, nil, http.MethodPut, *parsedRequestURL, bytes.NewReader(filePart))
```
Comment (lvirbalas) [inline govcd/upload.go]: Just noticed the typo in the func name "Wit", but probably it's better not to touch this public function at this late point in the release.

Comment (vbauzysvmware) [reply govcd/upload.go]: Oh nice, spot :) We can live with that for a while :-D this internal public function

**Comment by lroellin** [comment]
How can I use this latest version in Terraform?

**Comment by vbauzysvmware** [comment]
> How can I use this latest version in Terraform?

You would need to checkout terraform vDC provider and in go.mod reference my github branch.
Then build it and install it. More information is in REAMDE.

Or I could send you binary by some way.

**Comment by lroellin** [comment]
Yes, a binary would be much appreciated. I've got OSX 10.14.4

Btw. I had a look in go.mod: `github.com/vmware/go-vcloud-director/v2 v2.2.0-alpha.3`

What is the `v2` part of the URL for? I didn't find the v2.2... branch on the normal directory :)

**Comment by lroellin** [comment]
Sorry, had a better reading of the README now. I've built it and `make install`ed it. I'll check now!

**Comment by vbauzysvmware** [comment]
> Yes, a binary would be much appreciated. I've got OSX 10.14.4
> 
> Btw. I had a look in go.mod: `github.com/vmware/go-vcloud-director/v2 v2.2.0-alpha.3`
> 
> What is the `v2` part of the URL for? I didn't find the v2.2... branch on the normal directory :)

Binary:
https://we.tl/t-AhV7zQkig1

**Comment by lroellin** [comment]
I did get it to build now and it's installed (`make install`). I gave it a new version (2.1.5) and enforced it in the provider version. (`~> 2.1.5`)

Bad news: didn't work yet
```
PUT https://vcdXYZ.com/transfer/1161f228-7690-43ec-9922-f85e4939c5a8/file HTTP/1.1
Host: vcdXYZ.com
User-Agent: Go-http-client/1.1
Content-Length: 387072
Content-Range: bytes 0-387071/387072
Accept-Encoding: gzip
```

**Comment by lroellin** [comment]
Ok, problem solved. Probably didn't use the correct version.

With your binary (terraform-provider-vcd_v2.2.0-alpha4) it worked flawlessly!

**Comment by vbauzysvmware** [comment]
> Ok, problem solved. Probably didn't use the correct version.
> 
> With your binary (terraform-provider-vcd_v2.2.0-alpha4) it worked flawlessly!

Great. Thank you for verifying!!

**Pull request merged**